proximal/tests/_test_algs.py

```diff
@@ -2,7 +2,6 @@
 import proximal as px
 from proximal.lin_ops.vstack import vstack
 from proximal.algorithms import admm, pc, hqs, ladmm, absorb_offset
-from proximal.utils.cuda_codegen import PyCudaAdapter
 import cvxpy as cvx
 import numpy as np
```

Comment:
Why is the file renamed?

Reply:
A short answer is that it is work-in-progress, as stated in the commit message:

> * Rewrite test scripts in the folder proximal/tests to run via Pytest.
> Test scripts that are work-in-progress are renamed to `_test_*.py`

If you prefer a more elegant way to disable faulty tests, I can do it by following the suggestion in the link below:
https://stackoverflow.com/a/38442948/706334

The long answer is that the test class `TestAlgs` failed in the tip-of-tree, i.e. PR #47 . The tests complained that solvers in `proximal.algorithms.*` failed to reach global minimum. I have posted below the link to the error log:

https://pastebin.com/wFUpEniu

 I am not sure how and why the tests failed since year 2017; those example scripts `proximal/example/test_deconv.py` seemed to work quite well in denoising & deconvolving the simulated camera shake:
![test_deconv](https://user-images.githubusercontent.com/12185630/96289446-f1701700-0f99-11eb-8b72-f2b84357da0f.png)

Reply:
I see, sorry for not reading the commit message more carefully. Disabling tests is fine.

Reply:
I moved forward to revert the filenames back to `test_*.py`. And instead, I skipped the tests via the following approach:

```python
@pytest.mark.skipif('pycuda' if not in sys.modules, reason='Not supported')
def test_function(self):
   ...
```

And here is the expected result:
```
$ pytest proximal/tests
=== test session starts ===
platform linux -- Python 3.6.9, pytest-5.4.3, py-1.9.0, pluggy-0.13.1
rootdir: /home/antony/Documents/ProxImaL
plugins: cov-2.10.1
collected 54 items                                                                                                                                           

proximal/tests/test_algs.py .sssssssss                                                                                                                 [ 18%]
proximal/tests/test_cuda_comp_graph.py sssssssssssss                                                                                                   [ 42%]
proximal/tests/test_cuda_prox_fn.py s                                                                                                                  [ 44%]
proximal/tests/test_halide.py ...                                                                                                                      [ 50%]
proximal/tests/test_lin_ops.py ...........                                                                                                             [ 70%]
proximal/tests/test_problem.py ss..                                                                                                                    [ 77%]
proximal/tests/test_prox_fn.py ........                                                                                                                [ 92%]
proximal/tests/test_transforms.py s...                                                                                                                 [100%]

=== 28 passed, 26 skipped in 76.80s (0:01:16) ===
```